pandas/core/internals/concat.py

```diff
@@ -406,16 +408,14 @@ def is_na(self) -> bool:
             return all(isna_all(row) for row in values)
 
     @cache_readonly
-    def is_na_without_isna_all(self) -> bool:
+    def is_na_after_size_and_isna_all_deprecation(self) -> bool:
+        """
+        Will self.is_na be True after values.size == 0 deprecation and isna_all
+        deprecation are enforced?
+        """
         blk = self.block
         if blk.dtype.kind == "V":
             return True
-        if not blk._can_hold_na:
-            return False
-
-        values = blk.values
-        if values.size == 0:
-            return True
```

Comment:
why does this change?

Reply:
Because the future behavior won't depend on `values.size == 0` (note the changed method name/docstring)

Reply:
sure but the deprecation hasn't been enforced yet, why is this changing already?

Reply:
this method is for checking on the future behavior to see if we need to issue a warning.